### PR TITLE
Add ScriptServiceV3Alpha implementation

### PR DESCRIPTION
**Pull request by APErebus**
# Background

Following on from #635, this PR adds the implementation of `ScriptServiceV3Alpha`. This is a direct copy of the `ScriptServiceV2` class, with modifications to support the async methods.

# Results

I added the `IAsyncScriptServiceV3Alpha` in the same assembly as the implementation as it's not part of the "public" contract, just an internal contract.

I hooked up the new service into the `WorkspaceCleaner`.

I duplicated the `ScriptServiceV2Fixture` and `StartScriptCommandV2Builder` to verify that the new V3 implementation works identically to the existing V2 implementation

Note: It does mean any changes to `ScriptServiceV2` will need to be propagated to `V3Alpha`

Shortcut story: [sc-62810]

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [x] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle/Services/Scripts/ScriptServiceV3Alpha/ScriptServiceV3Alpha.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            var runningScript = runningScripts.GetOrAdd(
+                command.ScriptTicket,
+                _ =>
+                {
+                    var workspace = workspaceFactory.GetWorkspace(command.ScriptTicket);
+                    var scriptState = scriptStateStoreFactory.Create(workspace);
+                    return new RunningScriptWrapper(scriptState);
+                });
+
+            using (await runningScript.StartScriptMutex.LockAsync(cancellationToken))
```
Comment (APErebus) [inline source/Octopus.Tentacle/Services/Scripts/ScriptServiceV3Alpha/ScriptServiceV3Alpha.cs:46]: Changed this to async :)

File: source/Octopus.Tentacle/Services/Scripts/ScriptServiceV3Alpha/ScriptServiceV3Alpha.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+        public async Task CompleteScriptAsync(CompleteScriptCommandV3Alpha command, CancellationToken cancellationToken)
+        {
+            if (runningScripts.TryRemove(command.ScriptTicket, out var runningScript))
+            {
+                runningScript.Dispose();
+            }
+
+            var workspace = workspaceFactory.GetWorkspace(command.ScriptTicket);
+            await workspace.Delete(cancellationToken);
```
Comment (APErebus) [inline source/Octopus.Tentacle/Services/Scripts/ScriptServiceV3Alpha/ScriptServiceV3Alpha.cs:117]: Changed to async

File: source/Octopus.Tentacle/Services/Scripts/ScriptServiceV3Alpha/ScriptServiceV3Alpha.cs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        {
+            var runningScript = new RunningScript(shell, workspace, stateStore, workspace.CreateLog(), serverTaskId, cancellationToken, log);
+            var thread = new Thread(runningScript.Execute) { Name = "Executing PowerShell runningScript for " + ticket.TaskId };
+            thread.Start();
+            return runningScript;
+        }
+
+        async Task<ScriptStatusResponseV3Alpha> GetResponse(ScriptTicket ticket, long lastLogSequence, RunningScript? runningScript)
+        {
+            await Task.CompletedTask;
```
Comment (APErebus) [inline source/Octopus.Tentacle/Services/Scripts/ScriptServiceV3Alpha/ScriptServiceV3Alpha.cs:130]: Made this async as it's called in a bunch of places and it _will_ be async in the future :)

**Review by acodrington — APPROVED** [review]
Looks good.

Presumably there will be more PRs coming down the line to make SSv3Alpha available, but this is a good start!

**Comment by APErebus** [comment]
> Looks good.
> 
> Presumably there will be more PRs coming down the line to make SSv3Alpha available, but this is a good start!

Correct! The next PR (once this and #637 are merged) will be to expose V3Alpha and update TentacleClient to use it when available

**Pull request merged**